src/templatizer/dom-repeat.html

```diff
@@ -566,20 +567,23 @@
     // Called as a side effect of a host items.<key>.<path> path change,
     // responsible for notifying item.<path> changes to inst for key
     _forwardItemPath(path, value) {
-      if (this._itemsIdxToInstIdx) {
-        path = path.slice(6); // 'items.'.length == 6
-        var dot = path.indexOf('.');
-        var itemsIdx = dot < 0 ? path : path.substring(0, dot);
-        var instIdx = this._itemsIdxToInstIdx[itemsIdx];
-        var inst = this._instances[instIdx];
-        if (inst) {
-          path = dot < 0 ? '' : path.substring(dot+1);
-          if (!this._checkObservedPaths(path)) {
-            inst.forwardProperty(this.as + (path ? '.' + path : ''), value);
+      var itemsPath = path.slice(6); // 'items.'.length == 6
+      var dot = itemsPath.indexOf('.');
+      var itemsIdx = dot < 0 ? itemsPath : itemsPath.substring(0, dot);
+      // If path was index into array...
+      if (itemsIdx == parseInt(itemsIdx)) {
+        var itemPath = dot < 0 ? '' : itemsPath.substring(dot+1);
+        // See if the item subpath should trigger a full refresh...
+        if (!this._checkObservedPaths(itemPath)) {
+          // If not, forward to the instance for that index
+          var instIdx = this._itemsIdxToInstIdx[itemsIdx];
+          var inst = this._instances[instIdx];
+          if (inst) {
+            inst.forwardProperty(this.as + (itemPath ? '.' + itemPath : ''), value);
             inst.flushProperties();
-            return true;
-          }        
-        }
+          }
+        }        
+        return true;
```

Comment:
Rename to `handleItemPath`: returns true if item path was handled. Also rename `_checkObservedPaths` to `_handleObservedPaths` and give that the same contract.